content/en/docs/handling-errors.md

```diff
@@ -2,76 +2,157 @@
 title: "Handling Errors in the AWS SDK for Go V2"
 linkTitle: "Handling Errors"
 date: "2020-11-12"
-description: "Use the Error interface to handle errors from the |sdk-go| or AWS service."
+description: "Use the Error interface to handle errors from the AWS SDK for Go V2 or AWS service."
 ---
 
-The {{% alias sdk-go %}} returns errors that satisfy the Go `error`
-interface type and the `awserr.Error` interface in the
-``aws/awserr`` package. You can use the `Error()` method to get a formatted string of
-the SDK error message without any special handling.
+The {{% alias sdk-go %}} returns errors that satisfy the Go `error` interface type  You can use the `Error()` method to
+get a formatted string of the SDK error message without any special handling. Errors returned by the SDK may implement
+an `Unwrap` method. The `Unwrap` method is used by the SDK to provide additional contextual information to errors, while
+providing access to the underlying error or chain of errors. The `Unwrap` method should be used with the
+[errors.As](https://golang.org/pkg/errors#As) and [errors.Is](https://golang.org/pkg/errors#Is) functions which can
+handle unpacking wrapped error chains.
+
+It is important that your application check whether an error occurred after invoking a function or method that
+can return an `error` interface type. The most basic form of error handling looks similar to the following example:
+
+```go
+if err != nil {
+	// Handle error
+	return
+}
+```
+
+## Logging Errors
+
+The simplest form of error handling is traditionally to log or print the error message before returning or exiting from
+the application. For example:
+
+```go
+import "log"
+
+// ...
+
+if err != nil {
+	log.Printf("error: %s", err.Error())
+	return
+}
+```
+
+## Service Client Errors
+
+The SDK wraps All errors returned by service clients with the
+[smithy.OperationError]({{% apiref smithy="#OperationError" %}}) error type. `OperationError` provides contextual
+information about the service name and operation that is associated with an underlying error. This information can be
+useful for applications that perform batches of operations to one or more services, with a centralized error handling
+mechanism. Your application can use `errors.As` to access this `OperationError` metadata.
+
+For example:
+
+```go
+import "log"
+import "github.com/awslabs/smithy-go"
+
+// ...
+
+if err != nil {
+	var oe *smithy.OperationError
+	if errors.As(err, &oe) {
```

Comment:
`oe` doesn't need to be a pointer here

Reply:
This needs to be a pointer as OperationError implements the error interface as a pointer receiver type.

Reply:
That doesn't seem to be consistent with other usage then, eg for smithy.APIError